plotter/labels_test.go

```diff
@@ -5,12 +5,114 @@
 package plotter_test
 
 import (
+	"image/color"
 	"testing"
 
+	"gonum.org/v1/plot"
 	"gonum.org/v1/plot/cmpimg"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
 )
 
 func TestLabels(t *testing.T) {
 	cmpimg.CheckPlot(ExampleLabels, t, "labels.png")
 	cmpimg.CheckPlot(ExampleLabels_inCanvasCoordinates, t, "labels_cnv_coords.png")
 }
+
+func TestLabelsWithGlyphBoxes(t *testing.T) {
+	cmpimg.CheckPlot(
+		func() {
+			const fontSize = 24
+
+			p, err := plot.New()
+			if err != nil {
+				t.Fatalf("could not create plot: %+v", err)
+			}
+
+			p.Title.Text = "Labels"
+			p.X.Min = -1
+			p.X.Max = +1
+			p.Y.Min = -1
+			p.Y.Max = +1
+
+			labels, err := plotter.NewLabels(plotter.XYLabels{
+				XYs: []plotter.XY{
+					{X: -0.8 + 0.00, Y: -0.5},
+					{X: -0.6 + 0.02, Y: -0.5},
+					{X: -0.4 + 0.04, Y: -0.5},
+					{X: -0.8 + 0.00, Y: +0.5},
+					{X: -0.6 + 0.02, Y: +0.5},
+					{X: -0.4 + 0.04, Y: +0.5},
+					{X: +0.0 + 0.00, Y: +0},
+					{X: +0.2 + 0.02, Y: +0},
+					{X: +0.4 + 0.04, Y: +0},
```

Comment:
Can these have a note explaining why the non-simple form is being used, i.e. what the terms are.